lib/rubygems/request.rb

```diff
@@ -237,6 +237,14 @@ def perform_request(request) # :nodoc:
       verbose "fatal error"
 
       raise Gem::RemoteFetcher::FetchError.new('fatal error', @uri)
+    rescue (defined?(OpenSSL::SSL) ? OpenSSL::SSL::SSLError : nil) => e
```

Comment:
Why we rescuing for nil? if `OpenSSL::SSL` is not defined?

Reply:
I, possibly incorrectly, assumed that rescue was a form of `is_a?`.  It's been a few days since I did the PR, but I think I locally tested to see if it was catching other errors.

For the test to pass, it has to receive a `Gem::RemoteFetcher::FetchError` error, and if the CI passed for Ruby versions without TLS v1.3, nil is not catching other errors.